ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/server/BackupReplayFileTest.java

```diff
@@ -0,0 +1,150 @@
+
+    // When
+    replayFile.write(record.left, record.right);
+
+    // Then
+    final List<String> commands = Files.readAllLines(internalReplayFile.toPath());
+    assertThat(commands.size(), is(1));
+    assertThat(commands.get(0), is(
+        "\"stream/stream1/create\"" + KEY_VALUE_SEPARATOR
+            + "{\"statement\":\"CREATE STREAM stream1 (id INT) WITH (kafka_topic='stream1\"}"
+    ));
+  }
+
+  @Test
+  public void shouldWriteListOfRecordsTofile() throws IOException {
+    // Given
+    final Pair<CommandId, Command> record1 = newStreamRecord("stream1");
+    final Pair<CommandId, Command> record2 = newStreamRecord("stream2");
+
+    // When
+    replayFile.write(Arrays.asList(record1, record2));
+
+    // Then
+    final List<String> commands = Files.readAllLines(internalReplayFile.toPath());
+    assertThat(commands.size(), is(2));
+    assertThat(commands.get(0), is(
+        "\"stream/stream1/create\"" + KEY_VALUE_SEPARATOR
+            + "{\"statement\":\"CREATE STREAM stream1 (id INT) WITH (kafka_topic='stream1\"}"
+    ));
+    assertThat(commands.get(1), is(
+        "\"stream/stream2/create\"" + KEY_VALUE_SEPARATOR
+            + "{\"statement\":\"CREATE STREAM stream2 (id INT) WITH (kafka_topic='stream2\"}"
+    ));
+  }
+
+  @Test
+  public void shouldBeEmptyWhenReadAllCommandsFromEmptyFile() throws IOException {
+    // When
+    final List<?> commands = replayFile.readRecords();
+
+    // Then
+    assertThat(commands.size(), is(0));
+  }
+
+  @Test
+  public void shouldReadCommandsFromFile() throws IOException {
+    // Given
+    final Pair<CommandId, Command> record1 = newStreamRecord("stream1");
+    final Pair<CommandId, Command> record2 = newStreamRecord("stream2");
+    Files.write(internalReplayFile.toPath(),
+        String.format("%s%s%s%n%s%s%s",
+            "\"stream/stream1/create\"",
+            KEY_VALUE_SEPARATOR,
+            "{\"statement\":\"CREATE STREAM stream1 (id INT) WITH (kafka_topic='stream1\"}",
+            "\"stream/stream2/create\"",
+            KEY_VALUE_SEPARATOR,
+            "{\"statement\":\"CREATE STREAM stream2 (id INT) WITH (kafka_topic='stream2\"}"
+            ).getBytes(StandardCharsets.UTF_8));
+
+    // When
+    final List<Pair<CommandId, Command>> commands = replayFile.readRecords();
+
+    // Then
+    assertThat(commands.size(), is(2));
+    assertThat(commands.get(0).left, is(record1.left));
+    assertThat(commands.get(0).right, is(record1.right));
+    assertThat(commands.get(1).left, is(record2.left));
+    assertThat(commands.get(1).right, is(record2.right));
+  }
+
+  private Pair<CommandId, Command> newStreamRecord(final String streamName) {
+    final CommandId commandId = new CommandId(
+        CommandId.Type.STREAM, streamName, CommandId.Action.CREATE);
+    final Command command = new Command(
+        String.format("CREATE STREAM %s (id INT) WITH (kafka_topic='%s", streamName, streamName),
```

Comment:
let's add a test where the command has the newline character (e.g. `CREATE STREAM foo (id INT, "evil \n field" INT)...`)

Reply:
Done.